src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -264,5 +359,27 @@ internal static IEnumerable<Transaction> Order(IDictionary<AddressAsKey, Transac
         public bool SupportsBlobs => _transactionPool.SupportsBlobs;
 
         public override string ToString() => $"{nameof(TxPoolTxSource)}";
+
+        private class ArrayPoolBitMap : IDisposable
+        {
+            private const int BitShiftPerInt64 = 6;
+            private static int GetLengthOfBitLength(int n) => (n - 1 + (1 << BitShiftPerInt64)) >>> BitShiftPerInt64;
+
+            private readonly ulong[] _array;
+
+            public ArrayPoolBitMap(int size)
+            {
+                _array = ArrayPool<ulong>.Shared.Rent(GetLengthOfBitLength(size));
+                _array.AsSpan().Clear();
+            }
+
+            public bool this[int i]
+            {
+                get => (_array[i >> BitShiftPerInt64] & (1UL << i)) != 0;
+                set => _array[(uint)i >> BitShiftPerInt64] |= (1UL << i);
```

Comment:
This won't work for `false` unless you do the clearance first. I'd name it differently because in a month or two another person uses it without knowing that they can't `[i] = false` it. Paprika uses the first option https://github.com/NethermindEth/Paprika/blob/5792d74fe1596265059aecd8b017f139403df305/src/Paprika/Data/BitVector.cs#L216-L230